packages/social-auth/src/Auth.test.ts

```diff
@@ -52,50 +60,104 @@ describe("Auth", () => {
             const provider = new Provider(redirectSurface);
             jest.spyOn(provider, "login").mockImplementation(() =>
               Promise.resolve({
                 finalKeyData: { privKey: finalKey },
                 oAuthKeyData: { privKey: oAuthKey },
-                userInfo: [{ email, name }],
-              } as TorusAggregateLoginResponse)
+                userInfo: { email, name },
+              } as TorusLoginResponse)
             );
 
             await expect(provider.getCredentials()).resolves.toEqual({
               secretKey: resultKey,
               name: typeof resultName === "function" ? resultName(provider) : resultName,
             });
-          }
-        );
+          });
+        });
+      }
+    );
+  });
+
+  describe("getExtendedCredentials()", () => {
+    describe.each(testCases)(
+      "for $finalKey, $oAuthKey, $email, $name",
+      ({ finalKey, oAuthKey, email, name, profileImage, verifierId, resultKey }) => {
+        describe.each(redirectSurfaces)("for %s surface", redirectSurface => {
+          it.each([GoogleAuth, EmailAuth, RedditAuth])(
+            "handles aggregated login for %p",
+            async Provider => {
+              const provider = new Provider(redirectSurface);
+              jest.spyOn(provider, "login").mockImplementation(() =>
+                Promise.resolve({
+                  finalKeyData: { privKey: finalKey },
+                  oAuthKeyData: { privKey: oAuthKey },
+                  userInfo: [{ email, name, profileImage, verifierId }],
+                } as TorusAggregateLoginResponse)
+              );
 
-        it.each([FacebookAuth, TwitterAuth])("handles login for %p", async Provider => {
-          const provider = new Provider(redirectSurface);
-          jest.spyOn(provider, "login").mockImplementation(() =>
-            Promise.resolve({
-              finalKeyData: { privKey: finalKey },
-              oAuthKeyData: { privKey: oAuthKey },
-              userInfo: { email, name },
-            } as TorusLoginResponse)
+              await expect(provider.getExtendedCredentials()).resolves.toEqual({
+                secretKey: resultKey,
+                id: verifierId,
+                email,
+                name,
+                imageUrl: profileImage,
+              });
+            }
           );
 
-          await expect(provider.getCredentials()).resolves.toEqual({
-            secretKey: resultKey,
-            name: typeof resultName === "function" ? resultName(provider) : resultName,
+          it.each([FacebookAuth, TwitterAuth])("handles login for %p", async Provider => {
+            const provider = new Provider(redirectSurface);
+            jest.spyOn(provider, "login").mockImplementation(() =>
+              Promise.resolve({
+                finalKeyData: { privKey: finalKey },
+                oAuthKeyData: { privKey: oAuthKey },
+                userInfo: { email, name, profileImage, verifierId },
```

Comment:
did you actually check that all those providers give you such a response? the schema might be different

Reply:
Yep, checked for all providers we use, even if email is undefined, the field is still there. And the rest of the used fields are present